src/main/java/org/mapstruct/intellij/util/MapstructAnnotationUtils.java

```diff
@@ -314,7 +314,7 @@ else if ( mappingsValue instanceof PsiAnnotation ) {
 
         Stream<PsiAnnotation> mappingAnnotations = findMappingAnnotations( owner, includeMetaAnnotations );
 
-        return Stream.concat( mappingAnnotations, mappingsAnnotations );
+        return Stream.concat( mappingAnnotations, mappingsAnnotations ).distinct();
```

Comment:
Why is this needed? How can we have duplicates at this point?

Reply:
During working on this, I rewrote the logic a couple of times. Inside a debugging breakpoint I stumbled accross duplicates.
So this could be a leftover of a wrong implementation approach. But to be on the safe side, I will have another look.

But somewhere there must be a cause for duplicates. Because I had to change the logic of the candidate selection to a `HashSet` instead of an `ArrayList` ([here](https://github.com/mapstruct/mapstruct-idea/pull/152/files#diff-aad0b23f58d136bb1c125a22f4ddfcf83d07fe3b6ead490249a9e8465980b3c5R210)).

When I find the culprit, I will change it back again to an ArrayList.